gcore/src/exec.rs

```diff
@@ -202,9 +201,9 @@ pub fn reserve_gas(amount: u64, duration: u32) -> Result<ReservationId> {
 ///
 /// - [`reserve_gas`] function reserves gas for further usage.
 pub fn system_reserve_gas(amount: u64) -> Result<()> {
-    let mut len = 0u32;
-    unsafe { gsys::gr_system_reserve_gas(amount, &mut len as *mut u32) };
-    SyscallError(len).into_result()
+    let mut error_code = 0u32;
```

Comment:
As I can see in into_result implementation - we translate `0` code as Ok(()). But `0` is set by default in all cases. So, if by some reason node will not be able to write in `&error_code` memory chunk and by some unknown reasons program execution will not be stopped, then we will interpret this result as Ok(()). So, I think better to use some other default value.

Reply:
isn't it a memory error, if you can't mutate error code?

Reply:
0 is well-esablished success code, please keep it as it is